ql/src/java/org/apache/hadoop/hive/ql/hooks/HookContext.java

```diff
@@ -45,7 +47,50 @@
 public class HookContext {
 
   static public enum HookType {
-    PRE_EXEC_HOOK, POST_EXEC_HOOK, ON_FAILURE_HOOK
+
+    PRE_EXEC_HOOK(HiveConf.ConfVars.PREEXECHOOKS, ExecuteWithHookContext.class,
+        "Pre-execution hooks to be invoked for each statement"),
+    POST_EXEC_HOOK(HiveConf.ConfVars.POSTEXECHOOKS, ExecuteWithHookContext.class,
+        "Post-execution hooks to be invoked for each statement"),
+    ON_FAILURE_HOOK(HiveConf.ConfVars.ONFAILUREHOOKS, ExecuteWithHookContext.class,
+        "On-failure hooks to be invoked for each statement"),
+    QUERY_LIFETIME_HOOKS(HiveConf.ConfVars.HIVE_QUERY_LIFETIME_HOOKS, QueryLifeTimeHook.class,
+      "Hooks that will be triggered before/after query compilation and before/after query execution"),
+    SEMANTIC_ANALYZER_HOOK(HiveConf.ConfVars.SEMANTIC_ANALYZER_HOOK, HiveSemanticAnalyzerHook.class,
+      "Hooks that invoked before/after Hive performs its own semantic analysis on a statement"),
+    DRIVER_RUN_HOOKS(HiveConf.ConfVars.HIVE_DRIVER_RUN_HOOKS, HiveDriverRunHook.class,
+      "Hooks that Will be run at the beginning and end of Driver.run"),
+    REDACTOR(HiveConf.ConfVars.QUERYREDACTORHOOKS, Redactor.class,
+      "Hooks to be invoked for each query which can tranform the query before it's placed in the job.xml file"),
+    // The HiveSessionHook.class cannot access, use Hook.class instead
+    HIVE_SERVER2_SESSION_HOOK(HiveConf.ConfVars.HIVE_SERVER2_SESSION_HOOK, Hook.class,
+      "Hooks to be executed when session manager starts a new session"),
+    OOM(HiveConf.ConfVars.HIVE_SERVER2_OOM_HOOKS, Runnable.class,
```

Comment:
please keep some kind of naming contract between the varname and the enum key

Reply:
done, thank you